### PR TITLE
Task 2/work with date

### PR DESCRIPTION
**Pull request by KiriIl**

### CODE REVIEW

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            may = 31,
+            jun = 30,
+            jul = 31,
+            aug = 31,
+            sep = 30,
+            oct = 31,
+            nov = 30,
+            dec = 31
+        }
+        enum DaysOfWeek
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:24]: Please, keep empty lines between enums and methods

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            GetDayOfWeekMiddleLevel();
+            GetDayOfWeekHardLevel();
+        }
+
+        private static void GetDayOfWeekHardLevel()
+        {
+            Console.WriteLine("Введите дату ");
+            string dateStr = Console.ReadLine();
+            string[] strArr = dateStr.Split('.');
+            int dayInDate = int.Parse(strArr[0]), monthInDate = int.Parse(strArr[1]), yearInDate = int.Parse(strArr[2]), leapDays = CalcLeapDays(yearInDate);
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: It's better to not declare all these variables in the single line. One variable per line looks much better

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            int totalDays = 365 * (yearInDate - 1) + leapDays;
+            for (int i = 1; i < monthInDate; i++)
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:52]: Please, use `{`, `}`. If you add one more statement, you may forget to add brackets. This mistake seems to be stupid, but it's too frequent to don't care about it.

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            if (currentYearIsLeap)
+                leapDays--;
+            int everyCentury = yearInDate / 100;
+            int everyFourCentury = yearInDate / 400;
+            return leapDays - everyCentury + everyFourCentury;
+        }
+
+        private static bool Isleap(int yearInDate)
+        {
+            if (yearInDate % 4 == 0)
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: You can try to use `?:` operator here

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            switch(mod)
+            {
+                case 1: return DaysOfWeek.Monday;
+                case 2: return DaysOfWeek.Tuesday;
+                case 3: return DaysOfWeek.Wednsday;
+                case 4: return DaysOfWeek.Thursday;
+                case 5: return DaysOfWeek.Friday;
+                case 6: return DaysOfWeek.Saturday;
+                case 0: return DaysOfWeek.Sunday;
+                default: throw new Exception();
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: That's really nice👍
But never throw an exception without a message - exception should always reflect the reason why it has been thrown
```suggestion
                default: throw new Exception("Unexpected day of week");
```

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+            }
+        }
+
+        private static int GetDaysByMonth(int i, bool leapYear)
+        {
+            switch (i)
+            {
+                case 1: return (int)DaysInMonths.jan;
+                case 2 when leapYear: return (int)DaysInMonths.lFeb;
+                case 2 when !leapYear: return (int)DaysInMonths.Feb;
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:96]: 👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍.....👍👍👍👍

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+            DateTime date;
+            string inputString;
+            while (true)
+            {
+                Console.WriteLine("Введите дату ");
+                inputString = Console.ReadLine();
+                switch (inputString.ToLower())
+                {
+                    case "exit": return;
+                    default:
```
Comment (radzevich) [inline Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: `If-else` suits it better

**Review by radzevich — CHANGES_REQUESTED** [review]

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 1 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:24]: fixed

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 2 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: fixed

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 3 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:52]: fixed

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 4 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: remade

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 5 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: fixed

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 7 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs]: remade

File: Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs (hunk 6 above)
Comment (KiriIl) [reply Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs:96]: thanks)

**Review by radzevich — APPROVED** [review]

**Pull request merged**